protocol-units/da/movement/celestia/light-node/src/v1/passthrough.rs

```diff
@@ -65,34 +52,25 @@ where
 	}
 }
 
-impl<C> LightNodeV1Operations for LightNodeV1<C>
-where
-	C: PrimeCurve + CurveArithmetic + DigestPrimitive + PointCompression,
-	Scalar<C>: Invert<Output = CtOption<Scalar<C>>> + SignPrimitive<C>,
-	SignatureSize<C>: ArrayLength<u8>,
-	AffinePoint<C>: FromEncodedPoint<C> + ToEncodedPoint<C> + VerifyPrimitive<C>,
-	FieldBytesSize<C>: ModulusSize,
-{
+impl LightNodeV1Operations for LightNodeV1<LoadedSigner<Secp256k1>, Secp256k1> {
 	/// Tries to create a new LightNodeV1 instance from the toml config file.
 	async fn try_from_config(config: Config) -> Result<Self, anyhow::Error> {
 		let client = Arc::new(config.connect_celestia().await?);
 
-		let signing_key_str = config.da_signing_key();
-		let hex_bytes = hex::decode(signing_key_str)?;
-
-		let signing_key = SigningKey::from_bytes(hex_bytes.as_slice().try_into()?)
-			.map_err(|e| anyhow::anyhow!("Failed to create signing key: {}", e))?;
+		let loaded_signer: LoadedSigner<Secp256k1> =
+			<Config as LoadSigner<Secp256k1>>::da_signer(&config).await?;
+		let signer = Arc::new(Signer::new(loaded_signer));
 
 		Ok(Self {
 			config: config.clone(),
 			celestia_namespace: config.celestia_namespace(),
 			default_client: client.clone(),
-			verifier: Arc::new(Box::new(Verifier::<C>::new(
+			verifier: Arc::new(Box::new(Verifier::<Secp256k1>::new(
```

Comment:
... so this concrete implementation is responsible for hooking the verifier with the correct curve choice.